core/src/raw/http_util/client.rs

```diff
@@ -78,6 +81,34 @@ impl HttpClient {
 
     /// Fetch a request in async way.
     pub async fn fetch(&self, req: Request<Buffer>) -> Result<Response<HttpBody>> {
+        self.fetcher.fetch(req).await
+    }
+}
+
+#[async_trait::async_trait]
```

Comment:
Let's implement it in the same way as we do for `oio::Read`:

https://github.com/apache/opendal/blob/9f5f3ce00812b0a07e7648fac66bc4a01c1048d9/core/src/raw/oio/read/api.rs#L44-L105

We will have a `HttpFetch` and `HttpFetchDyn`